jina/drivers/cache.py

```diff
@@ -21,23 +21,20 @@ class BaseCacheDriver(BaseIndexDriver):
     def __init__(self, with_serialization: bool = False, *args, **kwargs):
         self.with_serialization = with_serialization
         super().__init__(*args, **kwargs)
-        self.field = None
 
     def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        self.field = self.exec.field
-
         if self._method_name == 'update':
-            self.exec_fn([d.id for d in docs], [d.id if self.field == ID_KEY else d.content_hash for d in docs])
+            values = [''.join(d.get_attrs(*self.exec.fields).values()) for d in docs]
+            self.exec_fn([d.id for d in docs],
+                         values)
```

Comment:
Interesting code cov says, the `self._method_name == 'update'`- case is not tested.
Is it right?

Reply:
It's tested here (mocked) : https://github.com/jina-ai/jina/blob/e6adf2b62e62255ab30f2c5b3e769bd79b7c985a/tests/unit/drivers/test_cache_driver.py#L206

and here: https://github.com/jina-ai/jina/blob/e6adf2b62e62255ab30f2c5b3e769bd79b7c985a/tests/integration/doccache/test_crud_cache.py#L203